server/lib/packets/login.go

```diff
@@ -3,44 +3,43 @@ package packets
 import (
 	"log"
 	
-	proto "../../proto"
-	"../models"
+	proto "github.com/umegaya/channer/server/proto"
+	"github.com/umegaya/channer/server/lib/models"
 )
 
 func ProcessLogin(from Source, msgid uint32, req *proto.LoginRequest, t Transport) {
 	user := req.User
 	version := req.Version
 	if AssetsConfig().App.ClientVersion != version {
 		log.Printf("login user:%v client outdated %v:%v", user, AssetsConfig().App.ClientVersion, version)
-		SendError(from, msgid, proto.Error_Login_OutdatedVersion)
+		SendError(from, msgid, &proto.Err{ Type: proto.Error_Login_OutdatedVersion })
 		return
 	}
 	rescue := req.Rescue
 	if rescue != nil {
 		a, err := models.FindRescueAccount(*rescue)
 		if err != nil {
-			SendError(from, msgid, proto.Error_Rescue_CannotRescue)
+			SendError(from, msgid, &proto.Err{ Type: proto.Error_Rescue_CannotRescue })
 			return
 		}
-		tmp := a.StringId()
 		req.User = a.User
 		req.Mail = a.Mail
-		req.Id = &tmp
+		req.Id = &a.Id
 		req.Pass = &a.Pass 
-		log.Printf("account %v rescue is enabled %v, force override secret", a.StringId(), *rescue)
+		log.Printf("account %v rescue is enabled %v, force override secret", a.Id, *rescue)
 	}
 	walltime := req.Walltime
 	//update account database
 	a, created, err := models.NewAccount(models.DBM(), req.Id, proto.Model_Account_User, user, req.Mail)
 	if err != nil {
 		log.Printf("login create account database error: %v", err)
-		SendError(from, msgid, proto.Error_Login_DatabaseError)
+		SendError(from, msgid, &proto.Err{ Type: proto.Error_Login_DatabaseError })
```

Comment:
__[gometalinter]__
* [gotype] undeclared name: proto
* [gotype] cannot pass argument &(proto.Err literal) (value of type *invalid type) to parameter of type error